python/lsst/dax/ppdb/bigquery/chunk_uploader.py

```diff
@@ -238,30 +241,44 @@ def _process_chunk(self, replica_chunk: PpdbReplicaChunkExtended) -> None:
 
         # Make a list of local parquet files to upload.
         parquet_files = list(chunk_dir.glob("*.parquet"))
-        if not parquet_files:
-            raise ChunkUploadError(chunk_id, f"No parquet files found in {chunk_dir}")
-
-        # Calculate total size of files to upload in bytes.
-        total_bytes = sum(p.stat().st_size for p in parquet_files)
 
-        uploads_completed = False
+        # Check if the chunk is expected to be empty.
+        is_empty = self._is_empty_chunk(manifest)
+
+        if not parquet_files and not is_empty:
+            # There is a mismatch between the manifest and the actual files.
+            # Some processing error may have occurred when exporting.
+            raise ChunkUploadError(chunk_id, f"No parquet files found in {chunk_dir} for non-empty chunk")
+
+        # Check that all expected parquet files from the manifest are present.
+        for table_name, table_stats in manifest.table_data.items():
+            if table_stats.row_count > 0:
+                expected_file = chunk_dir / f"{table_name}.parquet"
+                if not expected_file.exists():
```

Comment:
Just a random thought - I think this assumes that the local chunk cache is on a local file system? Would S3 be an option in the future? Maybe worth looking at `ResourcePath` for these operations?

Reply:
I like the suggestion but it is probably outside the scope of this PR to change. I will make a note of it on [DM-52326](https://rubinobs.atlassian.net/browse/DM-52326).